RecoTauTag/HLTProducers/src/CaloTowerFromL1TCreatorForTauHLT.cc

```diff
@@ -0,0 +1,107 @@
+  mEThreshold                                         (p.getParameter<double>       ("minimumE"          ) ),
+  mTauId                                              (p.getParameter<int>          ("TauId"             ) )
+{
+  produces<CaloTowerCollection>();
+}
+
+CaloTowerFromL1TCreatorForTauHLT::~CaloTowerFromL1TCreatorForTauHLT() {
+}
+
+void CaloTowerFromL1TCreatorForTauHLT::produce( StreamID sid, Event& evt, const EventSetup& stp ) const {
+  edm::Handle<CaloTowerCollection> caloTowers;
+  evt.getByToken( mtowers_token, caloTowers );
+
+  // imitate L1 seeds
+  edm::Handle<l1t::TauBxCollection> jetsgen;
+  evt.getByToken( mTauTrigger_token, jetsgen);
+
+  std::auto_ptr<CaloTowerCollection> cands( new CaloTowerCollection );
+  cands->reserve( caloTowers->size() );
+
+  int idTau = 0;
+  if (jetsgen.isValid()){ 
+    for (auto myL1Jet = jetsgen->begin(mBX); myL1Jet != jetsgen->end(mBX); myL1Jet++){      
+      if(idTau == mTauId){
+        double   Sum08 = 0.;
+        unsigned idx   = 0 ;
+        for (; idx < caloTowers->size(); idx++) {
+          const CaloTower* cal = &((*caloTowers) [idx]);
+          bool isAccepted = false;
+          if (mVerbose == 2) {
+              edm::LogInfo("JetDebugInfo") << "CaloTowerFromL1TCreatorForTauHLT::produce-> " << idx 
+                                           << " tower et/eta/phi/e: "                        << cal->et()  << '/' 
+                                                                                             << cal->eta() << '/' 
+                                                                                             << cal->phi() << '/' 
+                                                                                             << cal->energy() 
+                                           << " is...";
+              }
+          if (cal->et() >= mEtThreshold && cal->energy() >= mEThreshold ) {
+            math::PtEtaPhiELorentzVector p( cal->et(), cal->eta(), cal->phi(), cal->energy() );
+            double delta  = ROOT::Math::VectorUtil::DeltaR((*myL1Jet).p4().Vect(), p);
+            if(delta < mCone) {
+              isAccepted = true;
+              Sum08 += cal->et(); 
+              cands->push_back( *cal );
+            }
+          }
+          if (mVerbose == 2){
+              if (isAccepted) edm::LogInfo("JetDebugInfo") << "accepted \n";
+              else edm::LogInfo("JetDebugInfo") << "rejected \n";
+            }
+        }
+      }
+      idTau++;
+    }
+  } 
+  else {
+    edm::LogWarning("MissingProduct") << "L1Upgrade jet bx collection not found." << std::endl;
+  }
+  
+  evt.put( cands );
+  
+}
+
+void CaloTowerFromL1TCreatorForTauHLT::fillDescriptions( edm::ConfigurationDescriptions & desc ) {
+
+  edm::ParameterSetDescription aDesc;
+
+  aDesc.add<edm::InputTag>("TauTrigger"     , edm::InputTag("l1extraParticles","Tau"))->setComment("L1ExtraJet collection for seeding"                            );
```

Comment:
Sorry, I missed to update the comments and oversaw that the input tag is still set to l1extra.
I am updating the PR